Homework13/PageObjects/WebTablePage.cs

```diff
@@ -0,0 +1,126 @@
+        {
+            List<string> rowsText = new List<string>();
+            for (int i = 0; i < rows.Count; i++)
+            {
+                var stringBuilder = new StringBuilder();
+                {
+                    IWebElement element = rows[i];
+                    string text = element.Text;
+                    stringBuilder.Append($"{text} ");
+                }
+                var finalString = stringBuilder.ToString();
+                if (finalString != "        ")
+                {
+                    rowsText.Add(finalString);
+                }
+            }
+
+            return rowsText;
+        }
+
+        public int GetCountOfFilteredRows(List<string> rows, string filterValue) => rows.Count(x => x.Contains(filterValue));
+
+        public int GetCountOfRows(List<string> rows) => rows.Count;
+
+        public string GetTextOfCollectionElements(ReadOnlyCollection<IWebElement> selectedItems)
+        {
+            var stringBuilder = new StringBuilder();
+            for (int i = 0; i < selectedItems.Count; i++)
+            {
+                IWebElement element = selectedItems[i];
+                string text = element.Text;
+                stringBuilder.Append($"{text} ");
+            }
+            var finalString = stringBuilder.ToString();
+
+            return finalString;
+        }
+
+        public void FillInSearchInput(string value) => _searchInput.SendKeys(value);
+
+        public void FillInFirstNameInput(string value) => _firstNameInput.SendKeys(value);
+
+        public void FillInLastNameInput(string value) => _lastNameInput.SendKeys(value);
+
+        public void FillInEmailInput(string value) => _emailInput.SendKeys(value);
+
+        public void FillInAgeInput(string value) => _ageInput.SendKeys(value);
+
+        public void FillInSalaryInput(string value) => _salaryInput.SendKeys(value);
+
+        public void FillInDepartmentInput(string value) => _departmentInput.SendKeys(value);
+
+        public void ClearFirstNameInput() => _firstNameInput.SendKeys(Keys.Control + "a" + Keys.Delete);
+
+        public void ClearLastNameInput() => _lastNameInput.SendKeys(Keys.Control + "a" + Keys.Delete);
+
+        public void ClearEmailInput() => _emailInput.SendKeys(Keys.Control + "a" + Keys.Delete);
+
+        public void ClearAgeInput() => _ageInput.SendKeys(Keys.Control + "a" + Keys.Delete);
+
+        public void ClearSalaryInput() => _salaryInput.SendKeys(Keys.Control + "a" + Keys.Delete);
+
+        public void ClearDepartmentInput() => _departmentInput.SendKeys(Keys.Control + "a" + Keys.Delete);
+
+        public ReadOnlyCollection<IWebElement> GetNewRow()
+        {
+            var newRow = _row.FindElements(By.XPath("((//div[@role='row'])[5])/div"));
+
+            return newRow;
+        }
+
+        public ReadOnlyCollection<IWebElement> GetEditedRow()
+        {
+            var editedRow = _row.FindElements(By.XPath("((//div[@role='row'])[2])/div"));
```

Comment:
GetEditedRow - it returns only the first row, requires refactoring

Reply:
redone, pls check GetRow method

Reply:
redone, pls check GetRow method